File: javascript/3042.Count-Prefix-and-Suffix-Pairs-I.js
/**
 * @param {string[]} words
 * @return {number}
 */
var countPrefixSuffixPairs = function (words) {
  let count = 0;

  const isValid = (word1, word2) => {
    return word2.startsWith(word1) && word2.endsWith(word1);
  };

  for (let i = 0; i < words.length; i++) {
    for (let j = i + 1; j < words.length; j++) {
      if (isValid(words[i], words[j])) {
        count += 1;
      }
    }
  }

  return count;
};

File: python/3042.Count-Prefix-and-Suffix-Pairs-I.py
class Solution:
    def countPrefixSuffixPairs(self, words: List[str]) -> int:
        count = 0

        def isValid(word1, word2):
            return word2.startswith(word1) and word2.endswith(word1)

        for i in range(len(words)):
            for j in range(i + 1, len(words)):
                if isValid(words[i], words[j]):
                    count += 1

        return count